grails-app/views/inventory/list.gsp

```diff
@@ -72,6 +73,7 @@
                                 <g:each var="entry" in="${quantityMap.sort()}" status="i">
                                     <g:set var="product" value="${entry.key}"/>
                                     <g:set var="quantity" value="${entry.value}"/>
+                                    <g:set var="quantityAvailableToPromise" value="${entry?.key?.getQuantityAvailableToPromise(session.warehouse.id)}"/>
```

Comment:
a few problems here: 
1) This will cause a huge amount of queries (1 for every row) 
2) Database access should never occur within the view

For MVC applications, we always want the controller to gather the data, prepare the model, and pass the model to the view. The view just reads from the model and should not make calls to the service or data access layers. The only time where it's appropriate to do so would be when using a taglib that needs to access the database through the service. But this is not one of those times. 

Instead you want to create a model that has all the data we need. I think this could be done with a small refactoring where we drop the quantityMap and use a list of AvailableItems.